Non-deterministic results with random_state

Setting `random_state` appears to have no effect and results are non-deterministic.

```python
>>> import numpy as np
>>> from pynndescent import NNDescent

>>> seed = np.random.RandomState(42)

>>> x1 = seed.normal(0, 100, (1000, 50))
>>> x2 = seed.normal(0, 100, (1000, 50))

>>> index1 = NNDescent(x1, random_state=np.random.RandomState(42))
>>> neighbors1, distances1 = index1.query(x2)

>>> index2 = NNDescent(x1, random_state=np.random.RandomState(42))
>>> neighbors2, distances2 = index2.query(x2)

>>> np.testing.assert_equal(neighbors1, neighbors2)
AssertionError: Arrays are not equal

(mismatch 16.450000000000003%)
 x: array([867, 208, 495, ...,  75, 111, 222])
 y: array([867, 208, 495, ...,  75, 111,  91])

>>> np.testing.assert_equal(distances1, distances2)
AssertionError: Arrays are not equal

(mismatch 16.450000000000003%)
 x: array([747.84385 , 748.106632, 749.851705, ..., 741.814037, 743.790826,
       745.373141])
 y: array([747.84385 , 748.106632, 749.851705, ..., 741.814037, 743.790826,
       745.911667])
```

I created an empty conda environment with `python=3.6.7`. I installed pynndescent using `pip`:

```
> pip freeze

certifi==2018.11.29
llvmlite==0.26.0
numba==0.41.0
numpy==1.15.4
pynndescent==0.2.1
scikit-learn==0.20.1
scipy==1.1.0
```

--- a/pynndescent/tests/test_pynndescent_.py
+++ b/pynndescent/tests/test_pynndescent_.py
@@ -54,3 +54,19 @@ def test_angular_nn_descent_neighbor_accuracy():
         0.99,
         "NN-descent did not get 99% " "accuracy on nearest neighbors",
     )
+
+
+def test_deterministic():
+    seed = np.random.RandomState(42)
+
+    x1 = seed.normal(0, 100, (1000, 50))
+    x2 = seed.normal(0, 100, (1000, 50))
+
+    index1 = NNDescent(x1, random_state=np.random.RandomState(42))
+    neighbors1, distances1 = index1.query(x2)
+
+    index2 = NNDescent(x1, random_state=np.random.RandomState(42))
+    neighbors2, distances2 = index2.query(x2)
+
+    np.testing.assert_equal(neighbors1, neighbors2)
+    np.testing.assert_equal(distances1, distances2)

